libs/sdk-ui-ext/src/internal/utils/propertiesHelper.ts

```diff
@@ -243,3 +243,21 @@ export function getColumnWidthsFromProperties(
 ): ColumnWidthItem[] | undefined {
     return get(visualizationProperties, "controls.columnWidths");
 }
+
+export function getHighchartsChartConfiguration(
+    controlProperties: IVisualizationProperties,
+): IVisualizationProperties {
+    const isZoomInsight = get(controlProperties, "zoomInsight");
+    return isZoomInsight
+        ? {
+              zoomType: "x",
+              panKey: "shift",
+              panning: true,
+              resetZoomButton: {
+                  theme: {
+                      display: "none",
+                  },
+              },
+          }
+        : undefined;
+}
```

Comment:
this is misplaced. The 'zoomInsight' should be on IChartConfig as well and the creation of the highchart's zoom config that is part of 'chart' must be beyond our chart react component boundary.

this has two benefits:

-  code in plug viz does not need to omit the 'zoomInsight'; it can pass the zoomInsight down to underlying charts via IChartConfig

-  it is generally easier to reason when visualization properties content can be passed as is down to react component via IChartConfig (mapping is 1-1, public interface of plug viz is same as public interface of react components)

-  the users of react components can use 'zoomInsight' property in their config of the chart.

Note: the 'chart' prop is marked as 'internal' on IChartConfig (because is exposes chart implementation internals). we eventually want to get rid of it and have first-class public props to turn on various functionality.

Reply:
Thank you for your explanation, @lupko. I have updated the code, could you please review this again?